internal/components/components.go

```diff
@@ -143,7 +143,7 @@ func components() (component.Factories, error) {
 		udplogreceiver.NewFactory(),
 	}
 
-	receivers = append(receivers, extraReceivers()...)
```

Comment:
Where extraReceivers() was once called

Reply:
This isn't the same, and refers to the enabled unstable components if the build tag is set: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/ac315e8e921efeba1565a6dddec1f7c09fa00667/cmd/otelcontribcol/unstable_components_enabled.go

Reply:
Ahh, its made this way for components that have yet to merged into contrib correct? 

But because the unstable components files are apart of the main package, I won't have access to them when moving components.go to internal.

I think that I get what you are saying though, and that would be to move over the unstable_components files over to internal as well correct?

Reply:
They're for components that have been added to the contrib project (and are in the standard receiver, exporter, etc. directories) but haven't been added to the (previously) main component factories for vetting and reliability reasons.  By building a collector distribution with the [`unstable` tag](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/Makefile#L202) are they available.  Once hardened to a reasonable extent they get moved from the unstable_enabled location to the applicable slice.

Reply:
Oh wow ok that makes sense I was wondering about that unstable tag. I was thinking that makefiles were only used for testing purposes but it seems that it can also be used to help compilation maybe? Definitely going to look deeper into it but thanks for that explanation!